domain/secretbackend/state/state.go

```diff
@@ -46,18 +46,42 @@ func NewState(factory coredatabase.TxnRunnerFactory, logger Logger) *State {
 // GetModel is responsible for returning the model for the provided uuid. If no
 // model is found the given uuid then an error of
 // [github.com/juju/juju/domain/model/errors.NotFound] is returned.
-func (s *State) GetModel(ctx context.Context, uuid coremodel.UUID) (coremodel.Model, error) {
+func (s *State) GetModel(ctx context.Context, uuid coremodel.UUID) (*secretbackend.Model, error) {
```

Comment:
To return value over the pointer, yes, it's simpler and safer, but another thing we need to consider about is the data size because there is a performance difference.
And I don't think the value is always stored in the stack. It depends on how we will use the value in the caller(will it be `escaped`?).
But here, I'm happy to return the value.